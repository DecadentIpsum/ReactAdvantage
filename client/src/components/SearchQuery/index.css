.filters-wrapper {
  display: flex;
  justify-content: space-between;
  align-items: fex-start;
  border-bottom: 1px solid #B3B3AB; }
  @media (max-width: 640px) {
    .filters-wrapper {
      position: relative; } }

.search-query-wrapper {
  padding-bottom: 10px; }
  .search-query-wrapper > div > .condition {
    padding-left: 0; }
  .search-query-wrapper .condition {
    display: flex;
    align-items: center;
    flex-wrap: wrap;
    margin-left: 0;
    box-shadow: none; }
    @media (max-width: 640px) {
      .search-query-wrapper .condition {
        flex-direction: column;
        align-items: flex-start;
        justify-content: center; } }
  .search-query-wrapper .rule {
    box-shadow: none;
    flex-wrap: wrap; }
    @media (max-width: 640px) {
      .search-query-wrapper .rule {
        flex-direction: column;
        align-items: flex-start;
        justify-content: center; } }
  .search-query-wrapper .querySelect {
    margin-right: 10px;
    height: 26px;
    border-bottom: 1px solid #F5F5F3;
    color: #1E1E1E; }
    @media (max-width: 640px) {
      .search-query-wrapper .querySelect {
        margin-bottom: 10px;
        border-bottom-color: #B3B3AB; } }
    .search-query-wrapper .querySelect:focus {
      background-color: #F5F5F3; }
  .search-query-wrapper .ui-inputtext {
    height: 100%;
    box-sizing: border-box; }
  @media (max-width: 640px) {
    .search-query-wrapper .ui-button {
      margin-bottom: 10px; } }
  @media (max-width: 640px) {
    .search-query-wrapper .ui-button-danger {
      margin-top: 10px;
      margin-bottom: 0; } }

.search-query-title {
  font-size: 1rem;
  color: #1E1E1E; }

.filters {
  flex-basis: 80%; }

@media (max-width: 640px) {
  .filters-search-button-wrapper {
    position: absolute;
    top: 0;
    right: 0; } }
