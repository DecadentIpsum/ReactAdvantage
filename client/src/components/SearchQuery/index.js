import React, { Component } from 'react';
import TwoWayQuerybuilder from 'react-two-way-querybuilder-custom-select-fork';

import Button from 'components/Button';
import Dropdown from 'components/Dropdown';

import './index.css';

export default class SearchQuery extends Component {
    constructor(props) {
        super(props);
        this.config = {
            styles: {
                primaryBtn: 'ui-corner-all ui-widget ui-button ui-state-default',
                deleteBtn: 'ui-corner-all ui-widget ui-button ui-button-danger ui-state-default',
                rule: 'rule',
                condition: 'condition',
                select: 'querySelect',
                input: 'ui-inputtext ui-state-default ui-corner-all',
                txtArea: 'queryText',
            },
            selectRenderer: this.renderSelect,
        };
    }

    renderSelect = ({
        options,
        value,
        className,
        onChange,
    }) => (
        <Dropdown
            options={options}
            value={value}
            className="filters-dropdown"
            onChange={({ value }) => {
                onChange(value);
            }}
        />
    )

    renderButton() {
        const {
            onSearch,
            searchLabel = 'Search',
        } = this.props;

        return (
            <div className="filters-search-button-wrapper">
                <Button onClick={onSearch}>
                    {searchLabel}
                </Button>
            </div>
        );
    }

    render() {
        const {
            title = 'Filters',
            onSearch,
        } = this.props;

        return (
            <div className="filters-wrapper">
                <div className="filters">
                    <div className="search-query-wrapper">
                        {title && (
                            <div className="search-query-title">{title}</div>
                        )}
                        <div className="search-query">
                            <TwoWayQuerybuilder
                                fields={this.props.fields}
                                onChange={this.props.onFilterChange}
                                config={this.config}
                                {...this.props}
                            />
                        </div>
                    </div>
                </div>
                {onSearch && this.renderButton()}
            </div>
        );
    }
}
