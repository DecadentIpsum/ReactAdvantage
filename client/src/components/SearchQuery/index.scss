.filters-wrapper {
    display: flex;
    justify-content: space-between;
    align-items: fex-start;
    border-bottom: 1px solid #B3B3AB;

    @media(max-width: 640px) {
        position: relative;
    }
}

.search-query-wrapper {
    padding-bottom: 10px;

    > div > .condition {
        padding-left: 0;
    }

    .condition {
        display: flex;
        align-items: center;
        flex-wrap: wrap;
        margin-left: 0;
        box-shadow: none;

        @media (max-width: 640px) {
            flex-direction: column;
            align-items: flex-start;
            justify-content: center;
        }
    }

    .rule {
        box-shadow: none;
        flex-wrap: wrap;
        align-items: stretch;

        @media (max-width: 640px) {
            flex-direction: column;
            align-items: flex-start;
            justify-content: center;
        }
    }

    .querySelect {
        margin-right: 10px;
        height: 26px;
        border-bottom: 1px solid #F5F5F3;
        color: #1E1E1E;

        @media (max-width: 640px) {
            margin-bottom: 10px;
            border-bottom-color: #B3B3AB;
        }

        &:focus {
            background-color: #F5F5F3;
        }
    }

    .ui-inputtext {
        height: 100%;
        box-sizing: border-box;
    }

    .ui-button {
        @media (max-width: 640px) {
            margin-bottom: 10px;
        }
    }

    .ui-button-danger {
        @media (max-width: 640px) {
            margin-top: 10px;
            margin-bottom: 0;
        }
    }
}

.search-query-title {
    font-size: 1rem;
    color: #1E1E1E;
}

.filters {
    flex-basis: 80%;
}

.filters-search-button-wrapper {
    @media(max-width: 640px) {
        position: absolute;
        top: 0;
        right: 0;
    }
}

.filters-dropdown {
    margin-right: 0.25em;

    .ui-dropdown-panel,
    .ui-dropdown-items-wrapper {
        margin-left: 0;
    }
}
