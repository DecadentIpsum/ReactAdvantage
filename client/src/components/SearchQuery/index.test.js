import React from 'react';
import Enzyme, { shallow } from 'enzyme';

import SearchQuery from './index';

it('renders correctly', () => {
    const fields = [{
        name: 'test',
        operators: 'all',
        label: 'test',
        input: {
            type: 'text'
        }
    }];

    const component = shallow(
        <SearchQuery fields={fields} />
    );

    expect(component).toMatchSnapshot();
});