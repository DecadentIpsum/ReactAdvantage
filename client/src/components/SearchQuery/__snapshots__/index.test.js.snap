// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchQuery
    fields={
        Array [
            Object {
              "input": Object {
                "type": "text",
              },
              "label": "test",
              "name": "test",
              "operators": "all",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="filters"
>
          <div
                    className="search-query-wrapper"
          >
                    <div
                              className="search-query-title"
                    >
                              Filters
                    </div>
                    <div
                              className="search-query"
                    >
                              <TwoWayQuerybuilder
                                        buttonsText={Object {}}
                                        config={
                                                  Object {
                                                            "selectRenderer": [Function],
                                                            "styles": Object {
                                                              "condition": "condition",
                                                              "deleteBtn": "ui-corner-all ui-widget ui-button ui-button-danger ui-state-default",
                                                              "input": "ui-inputtext ui-state-default ui-corner-all",
                                                              "primaryBtn": "ui-corner-all ui-widget ui-button ui-state-default",
                                                              "rule": "rule",
                                                              "select": "querySelect",
                                                              "txtArea": "queryText",
                                                            },
                                                          }
                                        }
                                        fields={
                                                  Array [
                                                            Object {
                                                              "input": Object {
                                                                "type": "text",
                                                              },
                                                              "label": "test",
                                                              "name": "test",
                                                              "operators": "all",
                                                            },
                                                          ]
                                        }
                                        onChange={undefined}
                              />
                    </div>
          </div>
</div>,
        undefined,
      ],
      "className": "filters-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="search-query-wrapper"
>
            <div
                        className="search-query-title"
            >
                        Filters
            </div>
            <div
                        className="search-query"
            >
                        <TwoWayQuerybuilder
                                    buttonsText={Object {}}
                                    config={
                                                Object {
                                                            "selectRenderer": [Function],
                                                            "styles": Object {
                                                              "condition": "condition",
                                                              "deleteBtn": "ui-corner-all ui-widget ui-button ui-button-danger ui-state-default",
                                                              "input": "ui-inputtext ui-state-default ui-corner-all",
                                                              "primaryBtn": "ui-corner-all ui-widget ui-button ui-state-default",
                                                              "rule": "rule",
                                                              "select": "querySelect",
                                                              "txtArea": "queryText",
                                                            },
                                                          }
                                    }
                                    fields={
                                                Array [
                                                            Object {
                                                              "input": Object {
                                                                "type": "text",
                                                              },
                                                              "label": "test",
                                                              "name": "test",
                                                              "operators": "all",
                                                            },
                                                          ]
                                    }
                                    onChange={undefined}
                        />
            </div>
</div>,
          "className": "filters",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="search-query-title"
>
                Filters
</div>,
              <div
                className="search-query"
>
                <TwoWayQuerybuilder
                                buttonsText={Object {}}
                                config={
                                                Object {
                                                                "selectRenderer": [Function],
                                                                "styles": Object {
                                                                  "condition": "condition",
                                                                  "deleteBtn": "ui-corner-all ui-widget ui-button ui-button-danger ui-state-default",
                                                                  "input": "ui-inputtext ui-state-default ui-corner-all",
                                                                  "primaryBtn": "ui-corner-all ui-widget ui-button ui-state-default",
                                                                  "rule": "rule",
                                                                  "select": "querySelect",
                                                                  "txtArea": "queryText",
                                                                },
                                                              }
                                }
                                fields={
                                                Array [
                                                                Object {
                                                                  "input": Object {
                                                                    "type": "text",
                                                                  },
                                                                  "label": "test",
                                                                  "name": "test",
                                                                  "operators": "all",
                                                                },
                                                              ]
                                }
                                onChange={undefined}
                />
</div>,
            ],
            "className": "search-query-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Filters",
                "className": "search-query-title",
              },
              "ref": null,
              "rendered": "Filters",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <TwoWayQuerybuilder
                  buttonsText={Object {}}
                  config={
                                    Object {
                                                      "selectRenderer": [Function],
                                                      "styles": Object {
                                                        "condition": "condition",
                                                        "deleteBtn": "ui-corner-all ui-widget ui-button ui-button-danger ui-state-default",
                                                        "input": "ui-inputtext ui-state-default ui-corner-all",
                                                        "primaryBtn": "ui-corner-all ui-widget ui-button ui-state-default",
                                                        "rule": "rule",
                                                        "select": "querySelect",
                                                        "txtArea": "queryText",
                                                      },
                                                    }
                  }
                  fields={
                                    Array [
                                                      Object {
                                                        "input": Object {
                                                          "type": "text",
                                                        },
                                                        "label": "test",
                                                        "name": "test",
                                                        "operators": "all",
                                                      },
                                                    ]
                  }
                  onChange={undefined}
/>,
                "className": "search-query",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "buttonsText": Object {},
                  "config": Object {
                    "selectRenderer": [Function],
                    "styles": Object {
                      "condition": "condition",
                      "deleteBtn": "ui-corner-all ui-widget ui-button ui-button-danger ui-state-default",
                      "input": "ui-inputtext ui-state-default ui-corner-all",
                      "primaryBtn": "ui-corner-all ui-widget ui-button ui-state-default",
                      "rule": "rule",
                      "select": "querySelect",
                      "txtArea": "queryText",
                    },
                  },
                  "fields": Array [
                    Object {
                      "input": Object {
                        "type": "text",
                      },
                      "label": "test",
                      "name": "test",
                      "operators": "all",
                    },
                  ],
                  "onChange": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="filters"
>
            <div
                        className="search-query-wrapper"
            >
                        <div
                                    className="search-query-title"
                        >
                                    Filters
                        </div>
                        <div
                                    className="search-query"
                        >
                                    <TwoWayQuerybuilder
                                                buttonsText={Object {}}
                                                config={
                                                            Object {
                                                                        "selectRenderer": [Function],
                                                                        "styles": Object {
                                                                          "condition": "condition",
                                                                          "deleteBtn": "ui-corner-all ui-widget ui-button ui-button-danger ui-state-default",
                                                                          "input": "ui-inputtext ui-state-default ui-corner-all",
                                                                          "primaryBtn": "ui-corner-all ui-widget ui-button ui-state-default",
                                                                          "rule": "rule",
                                                                          "select": "querySelect",
                                                                          "txtArea": "queryText",
                                                                        },
                                                                      }
                                                }
                                                fields={
                                                            Array [
                                                                        Object {
                                                                          "input": Object {
                                                                            "type": "text",
                                                                          },
                                                                          "label": "test",
                                                                          "name": "test",
                                                                          "operators": "all",
                                                                        },
                                                                      ]
                                                }
                                                onChange={undefined}
                                    />
                        </div>
            </div>
</div>,
          undefined,
        ],
        "className": "filters-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="search-query-wrapper"
>
              <div
                            className="search-query-title"
              >
                            Filters
              </div>
              <div
                            className="search-query"
              >
                            <TwoWayQuerybuilder
                                          buttonsText={Object {}}
                                          config={
                                                        Object {
                                                                      "selectRenderer": [Function],
                                                                      "styles": Object {
                                                                        "condition": "condition",
                                                                        "deleteBtn": "ui-corner-all ui-widget ui-button ui-button-danger ui-state-default",
                                                                        "input": "ui-inputtext ui-state-default ui-corner-all",
                                                                        "primaryBtn": "ui-corner-all ui-widget ui-button ui-state-default",
                                                                        "rule": "rule",
                                                                        "select": "querySelect",
                                                                        "txtArea": "queryText",
                                                                      },
                                                                    }
                                          }
                                          fields={
                                                        Array [
                                                                      Object {
                                                                        "input": Object {
                                                                          "type": "text",
                                                                        },
                                                                        "label": "test",
                                                                        "name": "test",
                                                                        "operators": "all",
                                                                      },
                                                                    ]
                                          }
                                          onChange={undefined}
                            />
              </div>
</div>,
            "className": "filters",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="search-query-title"
>
                  Filters
</div>,
                <div
                  className="search-query"
>
                  <TwoWayQuerybuilder
                                    buttonsText={Object {}}
                                    config={
                                                      Object {
                                                                        "selectRenderer": [Function],
                                                                        "styles": Object {
                                                                          "condition": "condition",
                                                                          "deleteBtn": "ui-corner-all ui-widget ui-button ui-button-danger ui-state-default",
                                                                          "input": "ui-inputtext ui-state-default ui-corner-all",
                                                                          "primaryBtn": "ui-corner-all ui-widget ui-button ui-state-default",
                                                                          "rule": "rule",
                                                                          "select": "querySelect",
                                                                          "txtArea": "queryText",
                                                                        },
                                                                      }
                                    }
                                    fields={
                                                      Array [
                                                                        Object {
                                                                          "input": Object {
                                                                            "type": "text",
                                                                          },
                                                                          "label": "test",
                                                                          "name": "test",
                                                                          "operators": "all",
                                                                        },
                                                                      ]
                                    }
                                    onChange={undefined}
                  />
</div>,
              ],
              "className": "search-query-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Filters",
                  "className": "search-query-title",
                },
                "ref": null,
                "rendered": "Filters",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <TwoWayQuerybuilder
                    buttonsText={Object {}}
                    config={
                                        Object {
                                                            "selectRenderer": [Function],
                                                            "styles": Object {
                                                              "condition": "condition",
                                                              "deleteBtn": "ui-corner-all ui-widget ui-button ui-button-danger ui-state-default",
                                                              "input": "ui-inputtext ui-state-default ui-corner-all",
                                                              "primaryBtn": "ui-corner-all ui-widget ui-button ui-state-default",
                                                              "rule": "rule",
                                                              "select": "querySelect",
                                                              "txtArea": "queryText",
                                                            },
                                                          }
                    }
                    fields={
                                        Array [
                                                            Object {
                                                              "input": Object {
                                                                "type": "text",
                                                              },
                                                              "label": "test",
                                                              "name": "test",
                                                              "operators": "all",
                                                            },
                                                          ]
                    }
                    onChange={undefined}
/>,
                  "className": "search-query",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "buttonsText": Object {},
                    "config": Object {
                      "selectRenderer": [Function],
                      "styles": Object {
                        "condition": "condition",
                        "deleteBtn": "ui-corner-all ui-widget ui-button ui-button-danger ui-state-default",
                        "input": "ui-inputtext ui-state-default ui-corner-all",
                        "primaryBtn": "ui-corner-all ui-widget ui-button ui-state-default",
                        "rule": "rule",
                        "select": "querySelect",
                        "txtArea": "queryText",
                      },
                    },
                    "fields": Array [
                      Object {
                        "input": Object {
                          "type": "text",
                        },
                        "label": "test",
                        "name": "test",
                        "operators": "all",
                      },
                    ],
                    "onChange": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
